src/zelos/feeds.py

```diff
@@ -0,0 +1,226 @@
+    def __init__(
+        self, config, zml_parser: ZmlParser, hook_manager: HookManager
+    ):
+        self._hook_manager = hook_manager
+        self.logger = logging.getLogger(__name__)
+
+        # _inst_hook_info is only present when the inst feed is active.
+        self._inst_hook_info = None
+        # The syscall hook must be registered later
+        self._syscall_hook_info = None
+
+        self._feed_level = FeedLevel.NONE
+
+        self._handle_num = 0
+        self._subscribers: Dict[FeedLevel, Dict[int, Callable]] = {
+            FeedLevel.SYSCALL: {},
+            FeedLevel.API: {},
+            FeedLevel.INST: {},
+        }
+
+        # Default to feed level SYSCALL, unless plans for the syscall
+        # level feed were already specified.
+        if len(config.syscall_feed) == 0:
+            self.set_feed_level(FeedLevel.SYSCALL)
+
+        # For initial setup, we want to respect the highest level set if
+        # multiple are set to trigger immediately. We run them from
+        # lowest to highest to achieve this goal
+        for zml_string in config.stop_feed:
+            zml_parser.trigger_on_zml(
+                functools.partial(self.set_feed_level, FeedLevel.NONE),
+                zml_string,
+            )
+
+        for zml_string in config.api_feed:
+            zml_parser.trigger_on_zml(
+                functools.partial(self.set_feed_level, FeedLevel.API),
+                zml_string,
+            )
+
+        for zml_string in config.syscall_feed:
+            zml_parser.trigger_on_zml(
+                functools.partial(self.set_feed_level, FeedLevel.SYSCALL),
+                zml_string,
+            )
+
+        for zml_string in config.inst_feed:
+            zml_parser.trigger_on_zml(
+                functools.partial(self.set_feed_level, FeedLevel.INST),
+                zml_string,
+            )
+
+        # TODO: Turning on the syscall feed from a syscall hook skips
+        # the printing of the syscall it triggered on depening on the
+        # order of registration of the syscall hooks. (Of course,
+        # this could be fixed with a before hook)
+        self._syscall_hook_info = hook_manager.register_syscall_hook(
+            HookType.SYSCALL.AFTER,
+            self._syscall_feed_hook,
+            name="syscall_hook",
+        )
+
+    @property
+    def inst_feed_on(self) -> bool:
+        return self._feed_level >= FeedLevel.INST
+
+    @property
+    def api_feed_on(self) -> bool:
+        return self._feed_level >= FeedLevel.API
+
+    @property
+    def syscall_feed_on(self) -> bool:
+        return self._feed_level >= FeedLevel.SYSCALL
+
+    def get_feed_level(self) -> FeedLevel:
+        return self._feed_level
+
+    def set_feed_level(self, feed_level: FeedLevel):
+        self._feed_level = feed_level
+        self._refresh_inst_feed()
+        self._refresh_api_feed()
+        self._refresh_syscall_feed()
+
+    def subscribe_to_inst_feed(self, callback) -> FeedHandle:
```

Comment:
Can `Typing` be used to add the function prototype? The callback prototype should be specified somewhere here.